modules/core/src/main/java/org/apache/synapse/message/processor/impl/ScheduledMessageProcessor.java

```diff
@@ -243,8 +243,7 @@ public void setParameters(Map<String, Object> parameters) {
 	}
 
 
-    @Override
-    public boolean stop() {
+    public boolean stop(boolean isShuttingDown) {
```

Comment:
To me it looks like we need to have two methods
ScheduledMessageProcessor.stop()
ScheduledMessageProcessor.pause()

Then we can do what is necessary without passing isShuttingDown all the way down the methods? Of cause, if this parameter is useful for another, then it is fine. :)

Reply:
In general the isShuttingDown parameter was introduced to prevent deleting the Quartz task in a destroy if destroy is called in a graceful shut down, to allow another node to pick it up and execute it. This is required with the new implementation which uses Quartz's clustering features, which is based on sharing a database which holds the task data.

For ScheduledMessageProcessor specifically, the isShuttingDown parameter is used to NOT pause the task, if destroy is called in a graceful shut down, to allow another node to continue execution. If the task is paused in a shut down, another node will not execute it with the new implementation, since the task would be marked as paused in the shared database.